### PR TITLE
Final capstone Project Backend: Make changes to controllers and Model

### PR DESCRIPTION
**Pull request by binodbhusal**
**The following tasks were accomplished in the milestone**

- [x] Update motor controller resolved n+1 query problem
- [x] Update Reservation Controller 
- [x] Update Store Location Controller 
- [x] Added uniqueness to the city_name of the StoreLocation model
- [x] Update motor model added calculate_prices method
- [x] Added Store location seed data and other data

### CODE REVIEW

**Review by rahimialijan — APPROVED** [review]
Hi @binodbhusal 

## Status Approved
Your project is complete! There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉


## Highlights

1. Linter checking pass
2. Great file structured
3. New changes properly work with frontend

File: app/controllers/api/motors_controller.rb [hunk 1]
```diff
@@ -22,15 +22,28 @@ def new
...
+    city_name = params[:city_name]
 
-    @motor = @location.motors.new(motor_params)
-    @motor.user = current_user
+    # Find an existing StoreLocation by city_name
+    store_location = StoreLocation.find_by(city_name:)
+
+    if store_location.nil?
+      # If it doesn't exist, create a new StoreLocation
+      store_location = StoreLocation.create(city_name:)
```
Comment (dieum512) [inline app/controllers/api/motors_controller.rb:33]: Nice job updating the motors_controller :ok_hand: , But there is an issue you need to fix on line 29 and 33. it is giving this error when opening the motors API on the browser
Screenshot from 2023-10-07 21-06-13

Kindly, Udate the line 29 to this ``` store_location = StoreLocation.find_by(city_name: city_name) ``` and the line 33 to this ``` store_location = StoreLocation.create(city_name: city_name) ```. That will fix the error :wink:

**Review by dieum512 — CHANGES_REQUESTED** [review]
Hey @binodbhusal  

Your project is complete! There is nothing else to say other than... it's time to merge it :ok_hand: 
Congratulations! :tada: 



### HIGHLIGHTS
- [x] No  linter errors :heavy_check_mark: 
- [x] Good Commit messages :heavy_check_mark: 
- [x] Repository documented in a professional way :heavy_check_mark:

File: app/controllers/api/motors_controller.rb (hunk 1 above)
Comment (binodbhusal) [reply app/controllers/api/motors_controller.rb:33]: @dieum512 I have no error on my side I checked it. Did you seed the data? I will check again if any.

**Review by dieum512 — APPROVED** [review]

**Pull request merged**